### PR TITLE
Refactor `oe_get_param_size` and `oe_get_param_count`

### PR DESCRIPTION
**Pull request by andschwa**
Part of #1839, see commits.

### CODE REVIEW

**Comment by andschwa** [comment]
Ping @anakrish, I have two more PRs to open on top of this 😅

File: tools/oeedger8r/Emitter.ml [hunk 1]
```diff
@@ -126,55 +127,70 @@ let write_file (content : string list) (filename : string) (dir : string) =
...
+      else get_ptr_or_decl_size ptr_attr.pa_size decl
+  (* Values have no marshalling size. *)
   | _ -> ""
 
+(** For a parameter, get its count expression. *)
+let oe_get_param_count (ptype, decl, argstruct) =
+  let type_expr = get_type_expr ptype in
+  let get_ptr_or_decl_count (p : ptr_size) (d : declarator) =
+    match attr_value_to_string argstruct p.ps_count with
+    | Some s -> s
```
Comment (anakrish) [inline tools/oeedger8r/Emitter.ml]: Action item: Can you follow up if handling count & size scenario would require changing this function

File: tools/oeedger8r/Emitter.ml [hunk 2]
```diff
@@ -126,55 +127,70 @@ let write_file (content : string list) (filename : string) (dir : string) =
...
+    | None -> (
+      match attr_value_to_string argstruct p.ps_size with
+      | Some s -> sprintf "(%s / sizeof(%s))" s type_expr
+      | None ->
+          let dims = List.map string_of_int d.array_dims in
+          String.concat " * " dims )
+  in
+  match ptype with
+  | PTPtr (_, ptr_attr) ->
+      (* Strings are always a count of 1. *)
```
Comment (anakrish) [inline tools/oeedger8r/Emitter.ml]: This implementation reads odd..one would expect the count for the string to be strlen + 1.
Maybe you are planning to clean this up in next iteration?

File: tools/oeedger8r/Emitter.ml [hunk 3]
```diff
@@ -126,55 +127,68 @@ let write_file (content : string list) (filename : string) (dir : string) =
...
+  | None -> None
+  | Some (ANumber n) -> Some (string_of_int n)
+  | Some (AString s) -> Some (argstruct ^ s)
+
+(** For a parameter, get its size expression. *)
+let oe_get_param_size (ptype, decl, argstruct) =
+  let type_expr = get_type_expr ptype in
+  let get_ptr_or_decl_size (p : ptr_size) (d : declarator) =
+    match attr_value_to_string argstruct p.ps_count with
+    | Some s -> sprintf "(%s * sizeof(%s))" s type_expr
```
[the remark this answers is not in the document]
Comment (andschwa) [reply tools/oeedger8r/Emitter.ml]: Adding TODO. Really curious now why tests passed 😰

File: tools/oeedger8r/Emitter.ml (hunk 1 above)
Comment (andschwa) [reply tools/oeedger8r/Emitter.ml]: Ditto

File: tools/oeedger8r/Emitter.ml (hunk 2 above)
Comment (andschwa) [reply tools/oeedger8r/Emitter.ml]: Oh, heh, not 1 string, but count of chars in the string 😅

**Pull request merged**